From: Nedko Arnaudov <nedko@arnaudov.name>
Forwarded: not-needed
Origin: upstream, https://github.com/jackaudio/jack2/commit/e54729a300feb86e860df3553fcb135fa266a9c9
Bugs-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=675719
Applied-Upstream: 1.9.9
Description: Unbreak jack_control
--- a/example-clients/jack_control
+++ b/example-clients/jack_control
@@ -322,7 +322,7 @@
                 param = sys.argv[index]
                 index += 1
 
-                print "--- internal param reset \"%s\"" % (param
+                print "--- internal param reset \"%s\"" % param
 
                 configure_iface.ResetParameterValue(['internals', internal_name, param])
             elif arg == 'iload':
