From: Adrian Knoth <adi@drcomp.erfurt.thur.de>
Bugs-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=669566
Forwarded: Not-Yet
Description: Fix FTBFS on kFreeBSD
--- a/wscript
+++ b/wscript
@@ -79,10 +79,14 @@
     opt.sub_options('dbus')
 
 def configure(conf):
+    conf.load('compiler_cxx')
+    conf.load('compiler_cc')
     platform = sys.platform
     conf.env['IS_MACOSX'] = platform == 'darwin'
     conf.env['IS_LINUX'] = platform == 'linux' or platform == 'linux2' or platform == 'posix'
     conf.env['IS_SUN'] = platform == 'sunos'
+    if "kfreebsd" in platform:
+        conf.env['IS_LINUX'] = True
 
     if conf.env['IS_LINUX']:
         Logs.pprint('CYAN', "Linux detected")
