From: Adrian Knoth <adi@drcomp.erfurt.thur.de>
Forwarded: http://trac.jackaudio.org/ticket/264
Bugs-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651108
Description: Fix FTBFS on kFreeBSD
diff --git a/common/netjack.c b/common/netjack.c
index 075b97c..74cdc5d 100644
--- a/common/netjack.c
+++ b/common/netjack.c
@@ -47,7 +47,7 @@ $Id: net_driver.c,v 1.17 2006/04/16 20:16:10 torbenh Exp $
 #include <netinet/in.h>
 #endif
 
-#ifdef __linux__
+#if defined(HAVE_CONFIG_H)
 #include "config.h"
 #endif
 
diff --git a/common/netjack_packet.c b/common/netjack_packet.c
index 9434985..634a069 100644
--- a/common/netjack_packet.c
+++ b/common/netjack_packet.c
@@ -26,7 +26,7 @@
  *
  */
 
-#ifdef __linux__
+#if defined(HAVE_CONFIG_H)
 #include "config.h"
 #endif
 
diff --git a/example-clients/netsource.c b/example-clients/netsource.c
index 8da4d4f..e2b916e 100644
--- a/example-clients/netsource.c
+++ b/example-clients/netsource.c
@@ -34,7 +34,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 #include <string.h>
 #include <signal.h>
 
-#ifdef __linux__
+#if defined(HAVE_CONFIG_H)
 #include "config.h"
 #endif
 
