--- jackd2-1.9.10+20140719git3eb0ae6a~dfsg.orig/man/fill_template
+++ jackd2-1.9.10+20140719git3eb0ae6a~dfsg/man/fill_template
@@ -1,5 +1,5 @@
 #!/bin/sh
 
 for i in *.0 ; do
-   sed -e "s/!VERSION!/${1}/g" -e "s/!DATE!/`date '+%B %Y'`/g" < ${i} > ${i%%0}1
+   sed -e "s/!VERSION!/${1}/g" -e "s/!DATE!/`LC_ALL=C date --utc --date="$(dpkg-parsechangelog -l../debian/changelog -SDate)" '+%B %Y'`/g" < ${i} > ${i%%0}1
 done
